controllers/baseboardmanagement_controller.go

```diff
@@ -18,42 +18,222 @@ package controllers
+	err = bmcClient.SetPowerState(ctx, string(bm.Spec.Power))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// setCondition updates the status.Condition if the condition type is present.
+// Appends if new condition is found.
+// Patches the BaseboardManagement status.
+func (r *BaseboardManagementReconciler) setCondition(ctx context.Context, bm *bmcv1alpha1.BaseboardManagement, cType bmcv1alpha1.BaseboardManagementConditionType, message string) (ctrl.Result, error) {
+	patch := client.MergeFrom(bm.DeepCopy())
+
+	currentConditions := bm.Status.Conditions
+	for i := range currentConditions {
+		// If condition exists, update the message if different
+		if currentConditions[i].Type == cType {
+			if currentConditions[i].Message != message {
+				bm.Status.Conditions[i].Message = message
+				return r.patchStatus(ctx, bm, patch)
+			}
+			return ctrl.Result{}, nil
+		}
+	}
+
+	// Append new condition to Conditions
+	condition := bmcv1alpha1.BaseboardManagementCondition{
+		Type:    cType,
+		Message: message,
+	}
+	bm.Status.Conditions = append(bm.Status.Conditions, condition)
+
+	return r.patchStatus(ctx, bm, patch)
+}
+
+// reconcileStatus updates the Power and Conditions and patches BaseboardManagement status.
+func (r *BaseboardManagementReconciler) reconcileStatus(ctx context.Context, bm *bmcv1alpha1.BaseboardManagement) (ctrl.Result, error) {
+	patch := client.MergeFrom(bm.DeepCopy())
+
+	// Update the power status of the BaseboardManagement
+	bm.Status.Power = bm.Spec.Power
+	// Clear conditions
+	bm.Status.Conditions = []bmcv1alpha1.BaseboardManagementCondition{}
+
+	return r.patchStatus(ctx, bm, patch)
+}
+
+// patchStatus patches the specifies patch on the BaseboardManagement.
+func (r *BaseboardManagementReconciler) patchStatus(ctx context.Context, bm *bmcv1alpha1.BaseboardManagement, patch client.Patch) (ctrl.Result, error) {
+	logger := r.logger.WithValues("BaseboardManagement", bm.Name, "Namespace", bm.Namespace)
+
+	err := r.client.Status().Patch(ctx, bm, patch)
+	if err != nil {
+		logger.Error(err, "Failed to patch BaseboardManagement status")
+		return ctrl.Result{}, fmt.Errorf("failed to patch BaseboardManagement status: %v", err)
+	}
 
 	return ctrl.Result{}, nil
 }
 
+// resolveAuthSecretRef Gets the Secret from the SecretReference.
+// Returns the username and password encoded in the Secret.
+func (r *BaseboardManagementReconciler) resolveAuthSecretRef(ctx context.Context, secretRef corev1.SecretReference) (string, string, error) {
+	secret := &corev1.Secret{}
+	key := types.NamespacedName{Namespace: secretRef.Namespace, Name: secretRef.Name}
+
+	if err := r.client.Get(ctx, key, secret); err != nil {
+		if apierrors.IsNotFound(err) {
+			return "", "", fmt.Errorf("secret %s not found: %v", key, err)
+		}
+
+		return "", "", fmt.Errorf("failed to retrieve secret %s : %v", secretRef, err)
+	}
+
+	username, ok := secret.Data["username"]
+	if !ok {
+		return "", "", fmt.Errorf("required secret key username not present")
```

Comment:
I think this would be easier to understand if worded as `'username' required in BaseboardManagement secret: <baseboard management object name>`.

Same for password.